pandas/tests/groupby/test_categorical.py

```diff
@@ -963,3 +963,67 @@ def test_shift(fill_value):
                               categories=['a', 'b', 'c', 'd'], ordered=False)
     res = ct.shift(1, fill_value=fill_value)
     assert_equal(res, expected)
+
+
+def test_groupby_series_observed():
+    # GH 24880
+    df = DataFrame({'a': ['x', 'x', 'x', 'y'],
+                    'b': ['a', 'a', 'b', 'a'],
+                    'c': [1, 2, 3, 4]})
+    df['a'] = df['a'].astype('category')
+    df['b'] = df['b'].astype('category')
+
+    # test .agg and .apply when observed == False
+    lvls = [CategoricalIndex(['x', 'y'], categories=['x', 'y'], ordered=False),
+            CategoricalIndex(['a', 'b'], categories=['a', 'b'], ordered=False)]
+    index, _ = MultiIndex.from_product(lvls, names=['a', 'b']).sortlevel()
+    expected = pd.Series(data=[3, 3, 4, np.nan], index=index, name='c')
+    actual_agg = df.groupby(['a', 'b']).c.agg(sum)
+    actual_apply = df.groupby(['a', 'b']).c.apply(sum)
+    assert_series_equal(expected, actual_agg)
+    assert_series_equal(expected, actual_apply)
+
+    # test .agg when observed == True
+    index = MultiIndex.from_frame(df[['a', 'b']].drop_duplicates())
+    expected = pd.Series([3, 3, 4], index=index, name='c')
+    actual = df.groupby(['a', 'b'], observed=True).c.agg(sum)
+    assert_series_equal(expected, actual)
+
+    # test .apply when observed == True
+    index = MultiIndex.from_tuples([('x', 'a'), ('x', 'b'), ('y', 'a')],
+                                   names=('a', 'b'))
+    expected = pd.Series([3, 3, 4], index=index, name='c')
+    actual = df.groupby(['a', 'b'], observed=True).c.apply(sum)
+    assert_series_equal(expected, actual)
+
+
+def test_groupby_series_observed_apply_dict():
+    # GH 24880
+    df = DataFrame({'a': ['x', 'x', 'x', 'y'],
+                    'b': ['a', 'a', 'b', 'a'],
+                    'c': [1, 2, 3, 4]})
+    df['a'] = df['a'].astype('category')
+    df['b'] = df['b'].astype('category')
+
+    # observed == False
```

Comment:
Parametrize on True/False here as well. There should already be an observed fixture defined in the top level conftest.py that you can use

Reply:
I couldn't make use of the existing fixture due to other parameters. I also included the cases for None, in case the default changes. The tests look much cleaner now, but there's a couple of fixtures that probably won't be used often.